crates/sui-indexer-alt-graphql/src/api/types/gas_input.rs

```diff
@@ -63,12 +70,20 @@ impl GasInput {
         let limits = pagination.limits("GasInput", "gasPayment");
         let page = Page::from_params(limits, first, after, last, before)?;
 
-        let cursors = page.paginate_indices(self.native.payment.len());
+        // Filter out payments with zero object ID (sentinel values for system transactions)
+        let filtered_payments: Vec<_> = self
```

Comment:
Rather than filtering out the `ObjectID::ZERO`, check whether the sender is `SuiAddress::ZERO`, and if so, return an empty response.

The reason for this is that it's possible to have a user transaction show up with `0x0` as one of the payment coins (if you dry-run/simulate/execute the transaction), and in that case the filtering would be confusing. This also saves you from having to copy the payments field again in order to filter it.